Content.Server/StationEvents/StationEvent.cs

```diff
@@ -49,37 +91,104 @@ public abstract class StationEvent
             Occurrences += 1;
+        }
+
+        /// <summary>
+        /// Called when the tick is equal to the StartAfter variable.
+        /// </summary>
+        public abstract void Start();
+
+        /// <summary>
+        /// Called when after the time has elapsed to the AnnounceWhen variable.
+        /// </summary>
+        public virtual void Announce()
+        {
             if (StartAnnouncement != null)
             {
                 var chatManager = IoCManager.Resolve<IChatManager>();
                 chatManager.DispatchStationAnnouncement(StartAnnouncement);
             }
+            if (StartAudio != null)
+            {
+                EntitySystem.Get<AudioSystem>().PlayGlobal(StartAudio, AudioParams.Default.WithVolume(-10f));
+            }
+        }
+
+        /// <summary>
+        /// Called once when the station event ends
+        /// </summary>
+        public virtual void End()
+        {
+            if (EndAnnouncement != null)
+            {
+                var chatManager = IoCManager.Resolve<IChatManager>();
+                chatManager.DispatchStationAnnouncement(EndAnnouncement);
+            }
+            if (EndAudio != null)
+            {
+                EntitySystem.Get<AudioSystem>().PlayGlobal(EndAudio, AudioParams.Default.WithVolume(-10f));
+            }
+            Running = false;
         }
 
         /// <summary>
         /// Called every tick when this event is active
         /// </summary>
         /// <param name="frameTime"></param>
-        public abstract void Update(float frameTime);
+        public virtual void Tick(float frameTime)
+        {
+            return;
+        }
 
         /// <summary>
-        /// Called once when the station event ends
+        /// Called every tick when this event is running. Do not override me!
         /// </summary>
-        public virtual void Shutdown()
+        /// <param name="frameTime"></param>
+        public virtual void Update(float frameTime)
         {
-            if (EndAnnouncement != null)
+            if (!Running)
             {
-                var chatManager = IoCManager.Resolve<IChatManager>();
-                chatManager.DispatchStationAnnouncement(EndAnnouncement);
+                return;
+            }
+
+            if (ActiveFor >= StartAfter && !Started)
+            {
+                Start();
+                Started = true;
+            }
+
+            if (ActiveFor >= AnnounceWhen && !Announced)
+            {
+                Announce();
+                Announced = true;
+            }
+
+            if (StartAfter < ActiveFor && ActiveFor < EndWhen)
+            {
+                Tick(frameTime);
             }
+
+            if (ActiveFor >= EndWhen && ActiveFor >= AnnounceWhen && ActiveFor >= StartAfter)
+            {
+                End();
+                Started = false;
+                Announced = false;
+                return;
+            }
+            ActiveFor += frameTime;
```

Comment:
```suggestion
            Elapsed += frameTime;
         
            /* Have this called by StationEventSystem directly probably then you don't need to check it every tick?
            if (!Announced)
            {
                Announce();
                // Announced = true; Set this in Announce directly
            }         
            */
                  
            if (!Started && Elapsed >= StartAfter)
            {
                Startup();
                // Started = true; Set this in Startup directly
            }

            if (StartAfter < Elapsed)
            {
                Update(frameTime);
            }
```

Something like this, with the previous stuff at the end of Update being handled via End / Shutdown that also resets the variables etc.

Reply:
> Have this called by StationEventSystem directly probably then you don't need to check it every tick? 
we do not want it to tick unless it has started and not ended. Plus, sometimes we want to announce before starting it.

Reply:
Right but the system starts ticking it so it could Announce it when it starts and then never check again.